tensorflow_addons/losses/python/focal_loss.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.keras import backend as K
+
+
+def sigmoid_focal_crossentropy(y_true, y_pred, alpha=0.25,
```

Comment:
Could you implement it by a subclass of LossFunctionWrapper? And also, unit tests are necessary.

Example: [MeanSquaredError](https://github.com/tensorflow/tensorflow/blob/b85cb440e257a367fb70f8321ddaa669d1bd9fae/tensorflow/python/keras/losses.py#L169) and its test [MeanSquaredErrorTest](https://github.com/tensorflow/tensorflow/blob/b85cb440e257a367fb70f8321ddaa669d1bd9fae/tensorflow/python/keras/losses_test.py#L204)

Reply:
Sure. I will do both things. Thank you

Reply:
@facaiy is there any docker container for v2? I have installed it via pip on my local but it looks like some of the functionalities aren't there or should I install it again through pip only?

Reply:
We use tensorflow:nightly-custom-op image for addons. See developing section inhttps://github.com/tensorflow/addons/blob/master/README.md

Reply:
Thanks @facaiy . I have made the changes in the code and I am going to write the unit tests now. Though I have two doubts:
1) A unit tests should show that the value of `focal loss` must be higher when the confidence is low as compared to the loss for well classified examples. In order to check this I used this `losses_utils.ReductionV2.NONE` reduction, tested it locally by reducing on batch axis and keeping other dims same and everything is working fine.

2) If I am not wrong, any loss function should return a single scalar value in Keras. Hence I need to use some other reduction method. 

What do you suggest in this case?

Reply:
Hi @AakashKumarNain thanks for the PR. My answers for the questions:

1. The unit test should at the very least show that your loss function calculates an expected value when fed a test set of inputs. Classifying examples is outside of the scope of a loss unit test because that is dependent on the architecture that feeds the loss function.

1a. That being said... along with unit tests we highly encourage a full running example that shows in a [colab notebook](https://colab.research.google.com/) how this loss outperforms what was previously used (ideally the same metrics shown in a specific paper)

2. I think this is regarding the first method of the test class shown in @facaiy 's link? That was a test for the keras config, but all of the test methods in that class should be an example... not just one in particular.

Reply:
Thanks for the info @seanpmorgan.
As per your instructions, here is my implementation in colab with the tests. Please let me know what you think @facaiy @seanpmorgan 

https://colab.research.google.com/drive/1b4HLBvV2OKSK757tZ8Xy8n_Vxmw3umPj

Reply:
As originally suggested... can you make this loss a subclass of LossFunctionWrapper? This is a core concept of addons that all additions conform to a standard.